quadtree/src/lib.rs

```diff
@@ -55,11 +55,19 @@ pub struct Node {
 }
 
 impl Node {
-    pub fn root_with_bounding_rect(rect: Rect) -> Self {
-        Node {
-            id: NodeId::root(),
+    pub fn from_node_id_and_root_bounding_rect(id: NodeId, rect: Rect) -> Self {
+        let mut ids = vec![id];
```

Comment:
Maybe it would make sense to store only `child_indices` here instead of the whole `node_ids`? And store only the ids/child_indices of non-root nodes.

Reply:
Hm, I'm not sure whether this works. Currently, if the sub-root node is somewhere in the middle of the tree, we first need to find all nodes on the way to the root node. After finding these, we can use the `get_child()` method starting with the root node down to the sub-root node to compute the adjusted bounding box internally...